### DISCUSSION TITLE
Why is the [if available_contracts < count] check necessary in _build_contract_list()?

### DISCUSSION THREAD

**Discussion by SamJohn27149** [bug-or-feature/bc-utils#24] [Q&A] [answered]
Hi, 
I am new to bc-utile and PST. In the file 'bcutils/bc-utiles.py', i noticed the following logic:

```
if available_contracts < count:
    logger.warning(...)
    count = available_contracts

```

**My Understanding:**
    [count] and [available_contracts] are both generated from [contracts_per_instrument], I think [available_contracts] and [count] should always be equal in theory.
    
**My Questions:**
    Are there any scenarios where [available_contracts] < [count] can actually occur during runtime? If yes, what are those specific scenarios?

**Comment by bug-or-feature** [comment, marked as answer]
@SamJohn27149 this was the [pull request](https://github.com/bug-or-feature/bc-utils/pull/22/files). I never experienced the issue described myself. Is the change causing an issue?

**Reply by SamJohn27149** [reply]
Thanks for your reply.
I just found this logic difficult to understand(for me) when reading this code,  I haven't encountered any problems related to this in practice.
